stdlib/public/core/StringUTF8.swift

```diff
@@ -208,27 +208,46 @@ extension String {
     public struct Index : Comparable {
       internal typealias Buffer = _StringCore._UTF8Chunk
 
-      init(_ _core: _StringCore, _ _coreIndex: Int,
-           _ _buffer: Buffer) {
-        self._core = _core
+      init(_ _coreIndex: Int, _ _buffer: Buffer) {
         self._coreIndex = _coreIndex
         self._buffer = _buffer
-        _sanityCheck(_coreIndex >= 0)
-        _sanityCheck(_coreIndex <= _core.count)
+        // FIXME: move sanity checks elsewhere?
+        //_sanityCheck(_coreIndex >= 0)
+        //_sanityCheck(_coreIndex <= _core.count)
```

Comment:
I think it is safe to drop them.  Other code should have actual preconditions to ensure memory safety.  These checks were present to just discover library bugs earlier.